### ISSUE TITLE
`@rustbot ready` alias to switch to waiting on review?

### ISSUE THREAD

**Issue by joshtriplett** [rust-lang/triagebot#1339]
I would love to see simpler invocations that we can recommend to PR authors, that handle switching the various "waiting on" labels.

```
@rustbot ready
@rustbot waiting-on-author (or just "author")
```

Also, could we define sets of mutually exclusive labels, such as the various "waiting on" labels, so that you only have to add one and the bot will automatically remove others?

**Comment by jyn514** [comment]
> Also, could we define sets of mutually exclusive labels, such as the various "waiting on" labels, so that you only have to add one and the bot will automatically remove others?

I really want `S-waiting-on-help` for situations where *technically* it's waiting-on-author but realistically that won't happen without outside input.

**Comment by Mark-Simulacrum** [comment]
I at least only regularly look at waiting-on-review; any other state is likely to get entirely lost. I think it's reasonable to ask the reviewer to take point at either reassigning or identifying and pursuing that help.

**Comment by jyn514** [comment]
Duplicate of https://github.com/rust-lang/triagebot/issues/553.

**Comment by nikomatsakis** [comment]
I think this is a great idea. I find maintaining these labels pretty tedious. I'd be happy to help mentor someone here. I'm going to leave some notes.

**Comment by nikomatsakis** [comment]
I would also like the bot to pay attention to GH reviews. Here is what I would suggest:

* when the assigned reviewer leaves a GH review, if it is "changes requested", we should change the state of "waiting on author" (and leave a comment to that effect)
* add a `@rustbot ready` command to change the review to `S-waiting-on-reviewer` and probably add a comment pinging the assigned reviewer

The question is how easy/hard this will be to do =)

**Comment by nikomatsakis** [comment]
I'm thinking about the first part first -- the GH review with "changes requested". My *guess* is that this trigger a notification?

If so, I guess we want to edit this function:

https://github.com/rust-lang/triagebot/blob/67f108c7d5e6d808d16f5c29b2f4b9649af72e9d/src/handlers/notification.rs#L16

**Comment by Llandy3d** [comment]
After a first analysis, if we would want to support simpler single word abbreviations, from the example above `ready` & `author` for switching labels, we could create a `shortcut.rs` handler to keep it open for more additions down the line, what do you think ?

I also wonder if it would be better for the GH review section to be its own issue since the two are different in scope :thinking:

**Comment by nikomatsakis** [comment]
@Llandy3d that makes sense to me, yes

**Comment by nikomatsakis** [comment]
Just filed https://github.com/rust-lang/triagebot/issues/1373

**Comment by Mark-Simulacrum** [comment]
No, the handlers are per-function, that is, for this feature it likely makes sense to either:

* Add to the relabeling handler (I wouldn't do this)
* Create a new handler that handles the relevant events

It's likely that the github review is not too well integrated into our processing right now, so that may need some modifications to the github.rs file to parse the webhook events (at least adding some fields).

**Comment by kellda** [comment]
I think it has been fixed by #1381...
